cpg_infra/abstraction/azure.py

```diff
@@ -2,48 +2,187 @@
 """
 Azure implementation for abstract infrastructure
 """
+from datetime import date
+
+from typing import Any, Callable
 from functools import cached_property
-from typing import Any
 
 import pulumi_azure_native as az
 
+from cpg_infra.config import CPGInfrastructureConfig, CPGDatasetConfig
 from cpg_infra.abstraction.base import (
     CloudInfraBase,
     UNDELETE_PERIOD_IN_DAYS,
+    ARCHIVE_PERIOD_IN_DAYS,
     TMP_BUCKET_PERIOD_IN_DAYS,
     SecretMembership,
 )
-from cpg_infra.config import CPGInfrastructureConfig, CPGDatasetConfig
-
 
+AZURE_BILLING_START_DATE = "2017-06-01T00:00:00Z"
+AZURE_BILLING_EXPIRY_DATE = "3141-25-09T00:00:00Z"
 class AzureInfra(CloudInfraBase):
     def __init__(
         self, config: CPGInfrastructureConfig, dataset_config: CPGDatasetConfig
     ):
         super().__init__(config, dataset_config)
 
-        self.resource_group_name = f'{config.dataset_storage_prefix}{self.dataset}'
-        self.storage_account_name = f'{config.dataset_storage_prefix}{self.dataset}'
+        self.region = 'australiaeast'
```

Comment:
Can we make this configurable?

Reply:
We certainly can, but the gcp implementation currently doesn't either. Should we change both?